landlab/components/soil_grading/soil_grading.py

```diff
@@ -0,0 +1,540 @@
+        std: float, optional
+            The standard deviation of grain size distribution.
+        CV: float, optional
+            Coefficient of variance of the grain size distribution.
+        is_bedrock_distribution_flag: bool, optional
+            A flag to indicate if the grain size distribution is generated for soil or
+            bedrock layer.
+        precent_of_volume_in_spread: float, optional
+            The precent of volume transferred from parent to daughter in case of 'spread'
+             grading pattern
+        """
+
+        super().__init__(grid)
+
+        self._grading_name = grading_name
+        self._meansizes = meansizes
+        self._alpha = alpha
+        self._A_factor = A_factor
+
+        if self._meansizes is not None:
+            self._n_sizes = np.size(self._meansizes)
+            self._grain_max_size = np.max(self._meansizes)
+            self._input_sizes_flag = True
+
+        else:
+            self._input_sizes_flag = False
+            self._n_sizes = n_of_grainsize_classes
+            self._grain_max_size = grain_max_size
+
+        self._N = int(self._grading_name.split("-")[0][1:])
+        self._soil_density = float(soil_density)
+        self._phi = phi
+        self._power_of = power_of
+        self._initial_median_size = initial_median_size
+        self._CV = CV
+        self._initial_total_soil_weight = initial_total_soil_weight
+        if std is None:
+            std = self._initial_median_size * self._CV
+        self._std = std
+        self._is_bedrock_distribution_flag = is_bedrock_distribution_flag
+        self._precent_of_volume_in_spread = precent_of_volume_in_spread
+
+        # Note: Landlabs' init_out_field procedure will not work
+        # for the 'grains__weight' and 'grains_classes__size' fields
+        # because the shape of these fields is: n_nodes x n_grain_sizes.
+        grid.add_field(
+            "median_size__weight",
+            np.zeros(grid.shape[0] * grid.shape[1]),
+            at="node",
+            dtype=float,
+        )
+        grid.add_field(
+            "grains_classes__size",
+            np.ones((int(grid.shape[0] * grid.shape[1]), self._n_sizes)),
+            at="node",
+            dtype=float,
+        )
+        grid.add_field(
+            "grains__weight",
+            np.zeros((int(grid.shape[0] * grid.shape[1]), self._n_sizes)),
+            at="node",
+            dtype=float,
+        )
+        grid.add_field(
+            "bed_grains__proportions",
+            np.ones((int(grid.shape[0] * grid.shape[1]), self._n_sizes)),
+            at="node",
+            dtype=float,
+        )
+
+        try:
+            grid.add_field(
+                "soil__depth",
+                np.zeros(int(grid.shape[0] * grid.shape[1])),
+                at="node",
+                dtype=float,
+            )
+        except KeyError as exc:
+            raise ValueError("Soil field already exists") from exc
```

Comment:
Changed to: `grid.add_zeros("soil__depth", at="node")`.
Also, I changed the `KeyError` to `UserWarning` that will raise in case `soil__depth` field already exists (in this case, `soil__depth` will be overwritten by `SoilGrading`). 
Tests changed accordingly.

Reply:
Since the `soil__depth` field is an output field, it looks to me like you would want,
```python
if "soil__depth" not in grid.at_node:
    grid.add_empty("soil__depth", at="node")
grid.at_node["soil__depth"].fill(0.0)
```
There's no need to raise a warning—a user should expect an output field to be modified.

Reply:
Maybe it should be an `inout`, so a user could provide an already-populated field as an initial condition?

Reply:
I was thinking that too but then saw @yuvalsmilo's comment above https://github.com/landlab/landlab/pull/1946#discussion_r1660074058. It sounds as though the way to integrate a preexisting soil depth is to provide a grain size distribution and soil weight. `SoilGrading` then calculates the soil depth. I think.

Reply:
@gregtucker - yes, the user can init the component with a given grain size distribution, and then, the `soil__depth` field will be created accordingly. The case here is that `SoilGrading` component is initiated but a `soil__depth` field was already created for the grid (outside of `SoilGrading` ) and the match between grains mass and soil depth is not guarantee. In this case, `SoilGrading` will overwrite the `soil__depth` field to match grains mass.